controllers/signInAndOutController.js

```diff
@@ -69,14 +72,19 @@ module.exports = {
         // Otherwise correct user
         const userSession = new UserSession();
         userSession.userId = users._id;
-
         try {
             const doc = await userSession.save();
-            return res.send({
-                message: SUCCESSFUL_SIGNIN,
-                success: true,
-                token: doc._id,
-            });
+            await bcrypt.compare(password, users.password, ( err, result ) => {
+                if(result){
+                    return res.send({
+                        message: SUCCESSFUL_SIGNIN,
+                        success: true,
+                        token: doc._id,
+                    });
+                } else{
+                    return res.send({ err, message: INVALID_SIGNIN });
+                }
+            });           
```

Comment:
```
await bcrypt.compare(password, users.password);
return res.send({
...
});
```
the catch section is catch the error is the bcrypt fails to produce a match or there are other errors

Reply:
I think the catch sections, catches any error that does not result as a mismatch in password.
But the error parameter at the called back func is when there is false in the password match.

Reply:
is `bcrypt.compare` an async function?

Reply:
yes, it is